FWCore/Framework/interface/ESSourceDataProxyBase.h

```diff
@@ -53,19 +45,46 @@ namespace edm::eventsetup {
 
     virtual void prefetch(edm::eventsetup::DataKey const& iKey, EventSetupRecordDetails) = 0;
 
+    //Should call from prefetchAsyncImpl
+    template <typename ASYNC, typename GUARD>
+    void prefetchAsyncImplTemplate(ASYNC iAsync,
+                                   GUARD iGuard,
```

Comment:
How about
```suggestion
                                   GUARD iGuardFactory,
```
for more consistent naming with `guardPrefetch()`?

Reply:
done